### PR TITLE
Truncate log better

### PR DESCRIPTION
**Pull request by ppg**

### CODE REVIEW

File: log.go [hunk 1]
```diff
@@ -60,89 +65,120 @@ func SetMaxLogSize(logSize int64) {
...
-	rotateLog()
 	std.Critical(message)
 }
 
 func (log *Log) Error(message ...interface{}) {
-	if log.level <= ERROR {
+	log.rotateLog()
+
+	log.mu.RLock()
+	level := log.level
```
Comment (jamesbrown7) [inline log.go]: Can we break these three lines into their own function to avoid the repetition?

**Comment by jamesbrown7** [comment]
:+1: after minor refactor

File: log.go [hunk 2]
```diff
@@ -150,66 +166,106 @@ func Level(level int) {
...
+	return log.closeOutput()
 }
 
-func rotateLog() {
+// Not thread-safe
+func (log *Log) closeOutput() (err error) {
+	return log.logWriter.Close()
+}
 
+func CloseOutput() (err error) {
```
Comment (rmulley) [inline log.go]: Is this used anywhere? Seems like people will be using `log.CloseOutput()` only?

Comment (rmulley) [inline log.go]: NVM I see what's going on.

**Comment by rmulley** [comment]
:+1:

**Pull request merged**